src/js/game/LevelMVC/LevelPlane.js

```diff
@@ -524,11 +526,19 @@ module.exports = class LevelPlane {
     let armBlock = new LevelBlock(pistonType);
     for (let i = blocksPositions.length - 1; i >= 0; --i) {
       let destination = [blocksPositions[i][0] + offsetX, blocksPositions[i][1] + offsetY];
+      let block = this.getBlockAt(blocksPositions[i]);
+      if (this.getBlockAt(destination).blockType.startsWith("redstone") || this.getBlockAt(destination).blockType.startsWith("door")) {
```

Comment:
might as well go ahead and _use_ `getIsPushable` here :)

Reply:
In that case, we'll want a method for something like "getIsDestroyedWhenPushed" (which getIsPushable should probably use).

In general, we don't want stuff outside of LevelBlock to be making decisions by doing string comparisons on `blockType`; LevelBlock should be the only place that kind of logic lives

Reply:
👍